Front/dynamiccategory.h

```diff
@@ -0,0 +1,23 @@
+#ifndef DYNAMICCATEGORY_H
+#define DYNAMICCATEGORY_H
+
+#include <QPushButton>
+
+
+class DynamicCategory : public QPushButton
+{
+    Q_OBJECT
+public:
+    explicit DynamicCategory(QWidget *parent = 0);
+    ~DynamicCategory();
+    static int ResID;
```

Comment:
Possible race condition, if you would create a lot of `DynamicCategory` objects it can cause undefined behavior. Easiest solution is to use atomic variable here

Reply:
Fixed